haddock-api/src/Haddock.hs

```diff
@@ -461,7 +461,7 @@ render log' dflags unit_state flags sinceQual qual ifaces packages extSrcMap = d
             ppJsonIndex odir sourceUrls' opt_wiki_urls
                         unicode Nothing qual
                         ifaces
-                        ( nub
+                        ( ordNub
                         . map fst
                         . filter ((== Visible) . piVisibility . snd)
                         $ packages)
```

Comment:
This `nub` call in particular is troubling - `packages` is probably a very large list in some cases.